packages/langium/src/references/scope.ts

```diff
@@ -70,28 +90,42 @@ export class StreamScope implements Scope {
         }
         return undefined;
     }
+
+    getElements(name: string): Stream<AstNodeDescription> {
+        const lowerCaseName = this.caseInsensitive ? name.toLowerCase() : name;
+        const local = this.caseInsensitive
+            ? this.elements.filter(e => e.name.toLowerCase() === lowerCaseName)
+            : this.elements.filter(e => e.name === name);
+        if ((this.concatOuterScope || local.isEmpty()) && this.outerScope) {
+            return local.concat(this.outerScope.getElements(name));
+        } else {
+            return local;
+        }
+    }
 }
 
 export class MapScope implements Scope {
-    readonly elements: Map<string, AstNodeDescription>;
+    readonly elements: MultiMap<string, AstNodeDescription>;
```

Comment:
Shall we offer two different Scope implementations for Map and MultiMap? When performance is critical and there's no multi-reference, a Map might be better.

Reply:
I did some benchmarks, the `getElement` and `getElements` is roughly 5% slower, construction of the `MultiMap` roughly 20%. I kept `MapScope` and added `MultiMapScope` as a new implementation of `Scope`.